cas-server-core-web/src/main/java/org/jasig/cas/web/flow/AbstractNonInteractiveCredentialsAction.java

```diff
@@ -68,17 +78,21 @@ protected final Event doExecute(final RequestContext context) {
         final String ticketGrantingTicketId = WebUtils.getTicketGrantingTicketId(context);
         final Service service = WebUtils.getService(context);
 
-        if (isRenewPresent(context)
-            && ticketGrantingTicketId != null
-            && service != null) {
+        if (isRenewPresent(context) && ticketGrantingTicketId != null && service != null) {
 
             try {
+                final AuthenticationContextBuilder builder = new DefaultAuthenticationContextBuilder(
```

Comment:
I am personally in favor of this patch. While on the surface it might look
a bit overcompensated, in fact (IMHO) the design presents a well-defined,
cohesive abstractions that deal with authentication concern in a decoupled
manner and paves the way for a flexible a decoupled re-design of the web
SSO layer.

I could see that the factories objects here add to the "noise" and wonder
if those are absolute necessary at this time. But I'm +1 for now.

On Fri, Dec 18, 2015 at 04:14 Misagh Moayyed <notifications@github.com>
wrote:

> In
> cas-server-core-web/src/main/java/org/jasig/cas/web/flow/AbstractNonInteractiveCredentialsAction.java
> <https://github.com/Jasig/cas/pull/1343#discussion_r48007030>:
>
> >
> >              try {
> > +                final AuthenticationContextBuilder builder = new DefaultAuthenticationContextBuilder(
>
> The only other concept that remains unexplained here is
> authenticationObjectsRepository. That is meaningless to the story and
> it's just a technical glitch. Its main goal is to carry around objects that
> the authentication machinery needs, rather than injecting the same 3-4
> components into every class and action. It's like speed dial on your phone.
> :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jasig/cas/pull/1343/files#r48007030>.
>